### PR TITLE
Multiple issues fixed, see details

### PR DESCRIPTION
**Pull request by jasoncouture**
CancellationToken support (#166)
Timeout configuration support (#168)
~~Fixed project files for VS2017 RC as noted here: https://aka.ms/sdkimplicititems (#169)~~

There are a few changed API calls, any interface method where CancellationToken was present, but was not the last parameter was reordered to put CancellationToken last, and make it optional. The majority of methods are unaffected, but some such as ```Task GetContainerStatsAsync(string id, ContainerStatsParameters parameters, IProgress<JSONMessage> progress, CancellationToken cancellationToken = default(CancellationToken));``` were

I also made some parameters optional where there was default behavior for the call, I did not do this everywhere as I was not explcitly looking for it. The one example I changed was: LoadImageFromTarball, parameters was optional in the method, but not the parameters. So it was made optional.

### CODE REVIEW

**Comment by jasoncouture** [comment]
CLA was completed, please let me know if anything else is required.

**Comment by msftgits** [comment]
Hi, I am closing and re-opening this PR to bump the CLA bot. Sorry for the inconvenience!

File: src/Docker.DotNet/DockerClient.cs [hunk 1]
```diff
@@ -35,8 +35,10 @@ public sealed class DockerClient : IDockerClient
 
         public ISwarmOperations Swarm { get; }
 
+        public TimeSpan DefaultTimeout { get { return _defaultTimeout; } set { _defaultTimeout = value; } }
```
Comment (jterry75) [inline src/Docker.DotNet/DockerClient.cs]: Just make this an autoprop

```C#
public TimeSpan DefaultTimeout { get; set; }
```

and remove the _defaultTimeout

**Comment by jasoncouture** [comment]
Sure thing, I will take care of the merge conflicts and make that change.

File: src/Docker.DotNet/DockerClient.cs (hunk 1 above)
Comment (jasoncouture) [inline src/Docker.DotNet/DockerClient.cs]: Done in latest push. Also merged upstream and made sure the HTTP methods were correct from previous pull request.

**Comment by jterry75** [comment]
@jasoncouture - I fixed some conflicts. Let me know if that wasnt the fix you wanted.

**Comment by jterry75** [comment]
@jasoncouture - I dont understand what just happened. The merge seems exactly backwards of what it should be. You should be able to revert all *.csproj changes and then I can take this fix. We fixed the csproj's on a different pr.

**Comment by jasoncouture** [comment]
Will fix tomorrow at work. I will just revert back to commit  4d990eeb for the csproj files.

**Comment by jasoncouture** [comment]
@jterry75, fixed. Sorry it took so long, busy day at work. Just got around to it.
Also, excellent work thus far. This makes life so easy, before I was doing HTTP requests directly. So really, to all of you. Thank you. I'm glad I could contribute something.

File: src/Docker.DotNet/DockerClientConfiguration.cs [hunk 2]
```diff
@@ -8,12 +8,15 @@ public class DockerClientConfiguration : IDisposable
 
         public Credentials Credentials { get; internal set; }
 
-        public DockerClientConfiguration(Uri endpoint)
-            : this(endpoint, null)
-        {
-        }
+        public TimeSpan DefaultTimeout { get; internal set; } = TimeSpan.FromSeconds(100);
+
+        public DockerClientConfiguration(Uri endpoint, TimeSpan defaultTimeout) : this(endpoint, null, defaultTimeout) { }
```
Comment (jterry75) [inline src/Docker.DotNet/DockerClientConfiguration.cs]: Lets just do:

```C#
DockerClientConfiguration(Uri endpoint, Credentials credentials = null, TimeSpan defaultTimeout = default(TimeSpan))
{
Creds = creds ?? new Any();
if (defaultTimeout != TimeSpan.Zero)
DefaultTimeout = defaultTimeout;
}

**Comment by pauldotknopf** [comment]
This change has introduced a stackoverflow at ```MakeRequestAsync```.

**Pull request merged**